src/xdocs/checks/naming/abstractclassname.xml.template

```diff
@@ -0,0 +1,181 @@
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">
+  <head>
+    <title>AbstractClassName</title>
+  </head>
+  <body>
+    <section name="AbstractClassName">
+      <p>Since Checkstyle 3.2</p>
+      <subsection name="Description" id="Description">
+        <p>
+          Ensures that the names of abstract classes conforming to some pattern and
+          check that <code>abstract</code> modifier exists.
+        </p>
+        <p>
+          Rationale: Abstract classes are convenience base class implementations of
+          interfaces. For this reason, it should be made obvious that a given class
+          is abstract by prefacing the class name with 'Abstract'.
+        </p>
+      </subsection>
+
+      <subsection name="Properties" id="Properties">
+        <div class="wrapper">
+          <table>
+            <tr>
+              <th>name</th>
+              <th>description</th>
+              <th>type</th>
+              <th>default value</th>
+              <th>since</th>
+            </tr>
+            <tr>
+              <td>format</td>
+              <td>Specify valid identifiers.</td>
+              <td><a href="../../property_types.html#Pattern">Pattern</a></td>
+              <td><code>"^Abstract.+$"</code></td>
+              <td>3.2</td>
+            </tr>
+            <tr>
+              <td>ignoreModifier</td>
+              <td>
+                Control whether to ignore checking for the
+                <code>abstract</code> modifier on classes that match the
+                name.
+              </td>
+              <td><a href="../../property_types.html#boolean">boolean</a></td>
+              <td><code>false</code></td>
+              <td>5.3</td>
+            </tr>
+            <tr>
+              <td>ignoreName</td>
+              <td>
+                Control whether to ignore checking the name. Realistically
+                only useful if using the check to identify that match name
+                and do not have the <code>abstract</code> modifier.
+              </td>
+              <td><a href="../../property_types.html#boolean">boolean</a></td>
+              <td><code>false</code></td>
+              <td>5.3</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="Examples">
+        <p>
+          To configure the check:
+        </p>
+        <macro name="example">
+          <param name="path"
+                 value="src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java"/>
+          <param name="type"
+                 value="config"/>
+        </macro>
+        <p>Example:</p>
+        <macro name="example">
+          <param name="path"
+                 value="src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java"/>
```

Comment:
@rnveach @romani @nrmancuso Is providing the full path good or do we want to go shorter?

Reply:
If humans need to manage this, yes.

Reply:
Yes as in `Yes, it's good` or `Yes, we want to go shorter`?

Yes, people need to manage this.

Reply:
Make it easy for humans, everything up to `checks` should be the same for all examples right?

Reply:
Except `resources` and `resources-noncompilable`.

Reply:
@nrmancuso 
So something like this?
```bash
# imagine this is xml value: property
path: checks/mycheck/Example1.java
folder: resources-noncompilable
```

Or drop the `folder` and let the macro find it by trial and error?

Reply:
With simple naming like `Example1`, I do not think trial and error will work since we would presumably have a file named like this in both locations. Let's just go with everything after `xdocs-examples` for now, we can improve later if we come up with some good idea.

Reply:
> With simple naming like Example1, I do not think trial and error will work since we would presumably have a file named like this in both locations.

Why? I am trying to avoid this. The number at the end of the file name represents the order of that example in the xdoc(at least in my head) Moreover, for Java files that would be impossible because we cannot have two classes with the same name in a single package, right?

Reply:
Better to not make number in file name meaningful. Template define all, and can include any file we specify in it